File: Leetcode/Easy/two_sum.py
array = [3,4,5,7]
target = 11

def two_sum(array, target):
    map = {}
    for index, value in enumerate(array):
        # print(index, value)
        diff = target - value
        if diff in map:
            return (map[diff], index)
        map[value] = index
    return

_2sum = two_sum(array, target)
print(_2sum)

File: Leetcode/Easy/twoSum.js
array = [3,4,5,7]
target = 11

function twoSum(array, target){
    let map = {};
    for(let index = 0; index < array.length; index++){
        diff = target - array[index];
        if(map[diff] !== undefined){
            return [map[diff], index];
        } else {
            map[array[index]] = index;
        }
    }
    return
}

const _2sum = twoSum(array, target)
console.log(_2sum)